models/weight_recorder.cpp

```diff
@@ -155,16 +158,73 @@ nest::weight_recorder::set_status( const DictionaryDatum& d )
     user_set_precise_times_ = true;
 
   device_.set_status( d );
+
+  P_.set( d );
 }
 
+
 void
 nest::weight_recorder::handle( WeightRecorderEvent& e )
 {
   // accept spikes only if detector was active when spike was
   // emitted
   if ( device_.is_active( e.get_stamp() ) )
   {
+
+    // P_pre_ is defined and sender is not in it
+    if ( !P_.pre_.empty()
+      && std::find( P_.pre_.begin(), P_.pre_.end(), e.get_sender_gid() )
+        == P_.pre_.end() )
+      return;
```

Comment:
@weidel-p  I would suggest that you combine the pre and post tests. I think that will make the code more readable. Finally, we decided to write out logical operators in new NEST C++ code as in Python:
```C++
if ( ( not P_.pre_.empty() and std::find(...) == P_.pre_.end() )
      or ( not P_.post_.empty() and std::find(...) == P_.post_.end() ) ) 
```